sdk/core/azure-core/src/http/telemetry_policy.cpp

```diff
@@ -17,7 +17,47 @@
 
 #include <windows.h>
 
-#include "azure/core/internal/hkey_holder.hpp"
+#if !defined(WINAPI_PARTITION_DESKTOP) \
+    || WINAPI_PARTITION_DESKTOP // See azure/core/platform.hpp for explanation.
+
+namespace Azure { namespace Core { namespace _internal {
```

Comment:
@RickWinter what are your thoughts on the hkey_holder fix by moving the header into the only source that needs it? Specifically https://github.com/Azure/azure-sdk-for-cpp/pull/2188/commits/d713ea023bfb41d1a0b604afdfb5edadf023d079

Reply:
I think hkey_holder can be moved to sources.
And also I don't see you updating telemetry_policy that uses it.

Reply:
should you delete it, make sure you do
```
try
{
   handle = OpenHandle() // pseudocode
   ...
   CloseHandle() // pseudocode
}
catch(...) // literally 3 dots
{
   CloseHandle() // pseudocode
   throw; // literally just "throw;" without an exception
}
```

Reply:
> I think hkey_holder can be moved to sources.
And also I don't see you updating telemetry_policy that uses it.

I am confused. I did.

Reply:
Got it! You moved it here. This is fine. Maybe better to move it to a private (src) header. But not critical, to me.
In a separate header or not, should we remove other windows.h header usages or not, this is a good change.